pkg/cli/admin/upgrade/recommend/recommend.go

```diff
@@ -360,3 +362,63 @@ func findClusterOperatorStatusCondition(conditions []configv1.ClusterOperatorSta
 	}
 	return nil
 }
+
+func injectUpgradeableAsCondition(version string, condition *configv1.ClusterOperatorStatusCondition, majorMinorBuckets map[uint64]map[uint64][]configv1.ConditionalUpdate) error {
+	current, err := semver.Parse(version)
+	if err != nil {
+		return fmt.Errorf("cannot parse SemVer version %q: %v", version, err)
+	}
+
+	upgradeableURI := fmt.Sprintf("https://docs.openshift.com/container-platform/%d.%d/updating/preparing_for_updates/updating-cluster-prepare.html#cluster-upgradeable_updating-cluster-prepare", current.Major, current.Minor)
+	if current.Minor <= 13 {
+		upgradeableURI = fmt.Sprintf("https://docs.openshift.com/container-platform/%d.%d/updating/index.html#understanding_clusteroperator_conditiontypes_updating-clusters-overview", current.Major, current.Minor)
+	}
+
+	for major, minors := range majorMinorBuckets {
+		if major < current.Major {
+			continue
+		}
+
+		for minor, targets := range minors {
+			if major == current.Major && minor <= current.Minor {
+				continue
+			}
+
+			for _, target := range targets {
+				alreadyHasUpgradeableRisk := false
+				for _, risk := range target.Risks {
+					if strings.Contains(risk.Message, condition.Message) {
+						alreadyHasUpgradeableRisk = true
+						continue
+					}
+				}
+				if alreadyHasUpgradeableRisk {
+					continue
+				}
+
+				target.Risks = append(target.Risks, configv1.ConditionalUpdateRisk{
+					URL:           upgradeableURI,
+					Name:          "UpgradeableFalse",
+					Message:       condition.Message,
+					MatchingRules: []configv1.ClusterCondition{{Type: "Always"}},
+				})
+
+				for i, c := range target.Conditions {
+					if c.Type == "Recommended" {
+						if c.Status == metav1.ConditionTrue {
+							target.Conditions[i].Reason = condition.Reason
+							target.Conditions[i].Message = condition.Message
+						} else {
+							target.Conditions[i].Reason = "MultipleReasons"
+							target.Conditions[i].Message = fmt.Sprintf("%s\n\n%s", condition.Message, c.Message)
+						}
+						target.Conditions[i].Status = metav1.ConditionFalse
+						break
+					}
+				}
```

Comment:
Four levels of `for` depth makes this a bit hard to track at first, could be helpful to extract the inner block to something like:

```
if hasUpgradeableRisk(risks, message) {
	continue
}

addUpgradeableRisk(target, condition)
```

Reply:
refactored with 604071f5 -> 22cf5c6c, which also grows the ability to handle `target` that somehow have no `Recommended` condition at all.

/hold cancel